### PR TITLE
Philipye314/imp 2 qr code scanner

### PR DESCRIPTION
**Pull request by philipye314**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Created barcode scanner that reads information from a barcode and displays it on screen
### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
(SORRY IDK WHY ITS SO BIG)
IMG_4565

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
Open the QRCodeScanner subfolder in app/BottomTabNavigation/QRCodeScanner. The display of the camera and the crosshair is all dictated by CSS in the corresponding styles.ts file. I'd like some feedback on how exactly the crosshair should be displayed (I. e. should it be around the camera window? should the camera window take up the full screen, with the crosshair overlayed on top?). Should just be reviewed as a final commit. Haven't been able to test user permissions because expo never asks me for camera permission, but I wrote some basic logic to handle that -- not sure if that's part of my job. Otherwise the QRCodeScanner should be functional and you'll know it works if the dummy text below the crosshair updates.

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx [hunk 1]
```diff
@@ -1,11 +1,55 @@
-import React from 'react';
-import { Text, View } from 'react-native';
+import { BarCodeScanner, BarCodeScannerResult } from 'expo-barcode-scanner';
+import { router } from 'expo-router';
+import React, { useEffect, useState } from 'react';
+import { Image, Text, View } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx]: This is likely my fault but use `TouchableOpacity` from `react-native`, not `react-native-gesture-handler`

File: src/app/(BottomTabNavigation)/Cases/QRCodeScanner/img/crosshair.png
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/QRCodeScanner/img/crosshair.png:1]: We don't need the crosshair so let's remove this!

File: src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+    return <Text>CANNOT ACCESS CAMERA!</Text>;
+  }
+
   return (
     <View style={styles.container}>
-      <Text>This is the QR Code Scanner screen!</Text>
+      <Text>Add a new case</Text>
+      <BarCodeScanner
+        onBarCodeScanned={handleBarCodeScanned}
+        style={[styles.scanner]} // StyleSheet.absoluteFillObject,
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx]: Remove comment

File: src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx [hunk 3]
```diff
@@ -1,11 +1,55 @@
...
+      />
+      <Image style={styles.crosshair} source={require('./img/crosshair.png')} />
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx]: Let's remove the crosshair here. It's misaligned from my end for some reason but we also don't need!

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
So good! I think removing the `require()` statement should fix the eslint errors

File: src/app/(BottomTabNavigation)/Cases/QRCodeScanner/styles.ts [hunk 4]
```diff
@@ -7,4 +7,26 @@ export default StyleSheet.create({
...
+  },
+
+  button: {
+    margin: 10,
+    alignItems: 'center',
+    backgroundColor: '#339FFF',
+    padding: 10,
+  },
+
+  crosshair: {
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/QRCodeScanner/styles.ts]: Remove after removing crosshair!

File: src/screens/QRCodeScanner/QRCodeScanner.tsx
Comment (stephaniewong2) [inline src/screens/QRCodeScanner/QRCodeScanner.tsx:1]: Let's delete this and the `screens` directory

File: src/app/(BottomTabNavigation)/Cases/styles.ts [hunk 5]
```diff
@@ -7,4 +7,9 @@ export default StyleSheet.create({
     alignItems: 'center',
     justifyContent: 'center',
   },
+
+  scanner: {
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/styles.ts]: The `scanner` styles is located in `/src/app/(BottomTabNavigation)/Cases/QRCodeScanner/styles` so let's remove this

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Few minor comments but besides that looks good to me! Don't forget to pull from main because it'll block you from merging!

**Review by stephaniewong2 — APPROVED** [review]

**Comment by philipye314** [comment]
created working barcode scanner

**Pull request merged**